File: app/src/main/java/com/stoffe/gym/database/AppDatabase.java
package com.stoffe.gym.database;

import android.content.Context;

import com.stoffe.gym.database.Converters.LocalDateConverter;
import com.stoffe.gym.database.entities.Exercise;
import com.stoffe.gym.database.entities.ExerciseData;
import com.stoffe.gym.database.entities.Summary;
import com.stoffe.gym.database.entities.Workout;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {Workout.class, Exercise.class, ExerciseData.class, Summary.class},version = 9)
@TypeConverters({LocalDateConverter.class})
public abstract class AppDatabase extends RoomDatabase {

    public abstract WorkoutDao workoutDao();

    public static AppDatabase INSTANCE;

    public static AppDatabase getDatabase(final Context context){
        if(INSTANCE == null){
            synchronized (AppDatabase.class){
                if(INSTANCE == null){
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class,"person-databse").fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/stoffe/gym/database/AppDatabase.kt
package com.stoffe.gym.database

import android.content.Context
import androidx.room.AutoMigration
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.stoffe.gym.database.Converters.LocalDateConverter
import com.stoffe.gym.database.Converters.LocalDateTimeConverter
import com.stoffe.gym.database.entities.*

@Database(
    entities = [Workout::class, Exercise::class, ExerciseData::class, Summary::class, BMI::class],
    version = 10,
    exportSchema = true,
    autoMigrations = [
        AutoMigration(
            from = 9,
            to = 10,
        ),
    ]
)
@TypeConverters(LocalDateConverter::class, LocalDateTimeConverter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun workoutDao(): WorkoutDao
}

private lateinit var INSTANCE: AppDatabase

fun getDatabase(context: Context): AppDatabase {
    synchronized(AppDatabase::class.java) {
        if (!::INSTANCE.isInitialized) {
            INSTANCE = databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java, "person-databse"
            )
                .build()
        }
    }
    return INSTANCE
}
/*

private lateinit var INSTANCE: AppDatabase

fun getDatabase(context: Context): AppDatabase {
    synchronized(AppDatabase::class.java) {
        if (!::INSTANCE.isInitialized) {
            INSTANCE = Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "database-name"
            )
                .fallbackToDestructiveMigration()
                .build()
        }
    }
    return INSTANCE
}

 */